aptos-move/framework/supra-framework/sources/automation_registry_state.move

```diff
@@ -0,0 +1,226 @@
+            expiry_time,
+            max_gas_amount,
+            gas_price_cap,
+            is_active: false,
+            registration_epoch,
+            registration_time: timestamp::now_seconds(),
+            tx_hash,
+        };
+
+        enumerable_map::add_value(&mut registry_data.tasks, task_index, automation_task_metadata);
+        registry_data.current_index = registry_data.current_index + 1;
+        event::emit(automation_task_metadata);
+    }
+
+    /// Remove Automatioon task entry.
+    public (friend) fun remove_task(owner: &signer, id: u64): AutomationTaskMetaData acquires AutomationRegistryState {
+        let automation_registry = borrow_global_mut<AutomationRegistryState>(@supra_framework);
+        assert!(enumerable_map::contains(&automation_registry.tasks, id), EAUTOMATION_TASK_NOT_EXIST);
+
+        let automation_task_metadata = enumerable_map::get_value(&automation_registry.tasks, id);
+        assert!(automation_task_metadata.owner == signer::address_of(owner), EUNAUTHORIZED_TASK_OWNER);
+
+        enumerable_map::remove_value(&mut automation_registry.tasks, id);
+
+        // Adjust the gas committed for the next epoch by subtracting the gas amount of the expired task
+        automation_registry.gas_committed_for_next_epoch = automation_registry.gas_committed_for_next_epoch - automation_task_metadata.max_gas_amount;
+
+        event::emit(RemoveAutomationTask { id: automation_task_metadata.id });
+        // todo : return refund amount to user
+        automation_task_metadata
+    }
+
+    /// Update Automation gas limit
+    public (friend) fun update_automation_gas_limit(
+        supra_framework: &signer,
+        automation_gas_limit: u64
+    ) acquires AutomationRegistryState {
+        system_addresses::assert_supra_framework(supra_framework);
+
+        let automation_registry = borrow_global_mut<AutomationRegistryState>(@supra_framework);
+        automation_registry.automation_gas_limit = automation_gas_limit;
+
+        event::emit(UpdateAutomationGasLimit { automation_gas_limit });
+    }
+
+    /// List all the automation task ids
+    public(friend) fun get_active_task_ids(): vector<u64> acquires AutomationRegistryState {
+        let automation_registry = borrow_global<AutomationRegistryState>(@supra_framework);
+
+        let active_task_ids = vector[];
+        let ids = enumerable_map::get_map_list(&automation_registry.tasks);
+
+        vector::for_each(ids, |id| {
+            let task = enumerable_map::get_value(&automation_registry.tasks, id);
+            if (task.is_active) {
+                vector::push_back(&mut active_task_ids, id);
+            };
+        });
+        return active_task_ids
+    }
+
+    /// Retrieves the details of a automation task entry by its ID.
+    /// Returns a tuple where the first element indicates if the registry is completed/failed (`true`) or pending (`false`),
+    /// and the second element contains the `AutomationTaskMetaData` details.
```

Comment:
As I mentioned above I forgot to update the comment after making changes to the response type